vos/vos/tests/test_storage_inventory.py

```diff
@@ -0,0 +1,279 @@
+
+    @patch('vos.vos.md5_cache.MD5Cache.compute_md5')
+    @patch('__main__.open', MagicMock(), create=True)
+    def test_copy(self, computed_md5_mock):
+        # the md5sum of the file being copied
+        md5sum = 'd41d8cd98f00b204e9800998ecf84eee'
+        # patch the compute_md5 function in vos to return the above value
+        computed_md5_mock.return_value = md5sum
+
+        # mock the props of the corresponding node
+        props = MagicMock()
+        props.get.return_value = md5sum
+        # add props to the mocked node
+        node = MagicMock(spec=Node)
+        node.props = props
+
+        # mock one by one the chain of connection.session.response.headers
+        conn = MagicMock(spec=Connection)
+        session = MagicMock()
+        response = MagicMock()
+        mock_post_response = MagicMock(spec=requests.Response)
+        headers = {
+            'Location': 'https://gohere.com',
+            'content-disposition': 'mytestfiledownload.txt',
+            'Content-MD5': md5sum
+        }
+        response.headers = headers
+        session.get.return_value = response
+        response.status_code = 200
+        response.text = TRANSFER_XML
+
+        mock_post_response.status_code = 303
+        mock_post_response.headers = headers
+        session.post.return_value = mock_post_response
+
+        conn.session = session
+
+        test_client = Client(TestClient.TEST_SERVICE_RESOURCE_ID, conn=conn)
+
+        # Mock out the endpoints to avoid a true Registry service call.
+        nodes_url = 'https://ws-cadc.canfar.net/vault/nodes'
+        endpoints_mock = Mock()
+        endpoints_mock.nodes = nodes_url
+        test_client.get_endpoints = Mock(return_value=endpoints_mock)
+
+        mock_metadata_response = MagicMock()
+        mock_metadata_response.headers = {
+            'Content-Length': 88,
+            'Content-Type': 'application/fits',
+            'Content-MD5': 'd41d8cd98f00b204e9800998ecf8427e'
+        }
+
+        mock_ws_client = Mock(spec=net.BaseWsClient)
+        test_client._get_ws_client = Mock(return_value=mock_ws_client)
+        mock_ws_client.head.return_value = mock_metadata_response
+
+        # time to test...
+        storageLocation = 'cadc:TEST/foo'
+        osLocation = '/tmp/foo'
+        if os.path.isfile(osLocation):
+            os.remove(osLocation)
+        # copy from Storage Inventory
+        test_client.copy(storageLocation, osLocation)
+        computed_md5_mock.assert_called_once_with(osLocation)
+
+        # repeat - local file and vospace file are now the same -> only
+        # get_node is called to get the md5 of remote file
+        computed_md5_mock.reset_mock()
+        props.reset_mock()
+        props.get.return_value = md5sum
+        test_client.copy(storageLocation, osLocation)
+        computed_md5_mock.assert_called_once_with(osLocation)
+
+        # change the content of local files to trigger a new copy
+        # computed_md5_mock.reset_mock()
```

Comment:
Tests of the results is the same with the case above when copy was not triggered.

In general these tests seem to just test that the method returns successfully or not and not the actual behaviour.

Reply:
Added file size check.